crates/libs/bindgen/src/rust/winrt_methods.rs

```diff
@@ -47,38 +60,64 @@ pub fn writer(
         }
     };
 
+    let vcall = quote! { (windows_core::Interface::vtable(this).#vname)(windows_core::Interface::as_raw(this), #args #return_arg) };
+
     let vcall = match &signature.return_type {
         metadata::Type::Void => {
-            quote! {
-                (windows_core::Interface::vtable(this).#vname)(windows_core::Interface::as_raw(this), #args).ok()
+            if noexcept {
+                quote! {
+                    _ = #vcall;
```

Comment:
is it worth generating a `debug_assert` here to make sure the call actually returned `S_OK` like its contract says it should?

Reply:
done!